src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -107,6 +107,17 @@ private bool EmitNupkg(PublishRoot root)
                 root.Operations.Delete(TargetPath);
             }
 
+            // If this is a wrapper project, we need to generate a lock file before building it
+            if (IsWrappingAssembly())
+            {
+                var success = Restore(root, publishProject: this, restoreDirectory: project.ProjectDirectory)
```

Comment:
I'm worried about functional failures. What happens if I never restore my project references to csproj? It actually seems like this can happen for other `project.json` projects as well.